packages/application/src/requests.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+import { URLExt } from '@jupyterlab/coreutils';
+import { ServerConnection } from '@jupyterlab/services';
+
+import * as React from 'react';
+
+import { ErrorDialogContent } from './ErrorDialogContent';
+
+const HTML_STATUS_SUCCESS = [200, 201];
+
+export class RequestHandler {
```

Comment:
Can we make this visible only to application package, and expose this via Application.invokeService or something like that ?

Reply:
But this is meant to be a utility class, though I would agree that making all but `makePostRequest` and `makeGetRequest` private would be prudent. I'm unsure how we would make these two functions available outside of application otherwise.